### PR TITLE
Voluapp 1455 vol2obj v13

### PR DESCRIPTION
**Pull request by capnramses**
## Purpose of this PR

* New feature - vol2obj support for v1.3 volograms in a single file, with Basis Universal textures.
* New feature - check disk space before writing images. 
* Also checked `fwrite` calls for failure when writing .objs to catch disk space fails there.
* Bonus features: drag&drop support for single-file volograms.

## Changes made in this PR

* Fixed a portability bug in vol_geom.c on windows - and PRd to vol_libs.
* Included files for basis universal.
* Included vol_geom wrapper from vol_libs.
* Updated Makefile.
* Updated build_msvc.bat
* Added drag&drop support for single-file volograms as well as folder of older vologram files.

## Testing Summary

Provide more detail here as required.

- [x] Explained how reviewers build, deploy, and test your code and what should they look for that would indicate a bug or regression.
- [x] Tested locally. Windows and Ubuntu. **Need help testing on macOS!**
- [x] Stated supported platforms have you not tested on, that you need help with?
- [x] Named specific tools have you ran over the code: e.g. which { gcc Ubuntu, minGW, cl (msvc) compilers, GDB debugged with all warnings. No sanitiser or linter pass done. }.
- [x] Explained inputs used to test for bugs. e.g. existing known working data, large files, empty files, etc.
Tested on 2x v1.3 volograms, and 1x v1.2 vologram on Windows and Linux.

## Risk Introduced and any Breaking Changes

Provide more detail here as required.

- [ ] This PR introduces breaking changes (API or ABI). Instructions are provided for existing users to update their projects.
- [x] This PR introduces new dependencies. You have provided links and licences in this section. This is **_extremely important!_**. I updated the ReadMe with licence info copied from vol_libs.
- [x] New dependencies: Explained how any new dependencies are linked and built againt - .dll, inline code etc.? Makefile and build .bat handle this.
- [ ] Gaps in testing: Is any code not covered by tests?
- [ ] Gaps in testing: Are you unable to test on some platform? Or is some input data not available to you for testing?
- [x] Analysed any other potential risks here. I think the cutter tool is out of date and will require an update for v1.3 files and any changes to libraries. Not part of this ticket.

## Pre-Merge Checklist

* If you have a problem with the review process (e.g. it's taking too long, or the reviewers forgot) get in touch with the Team Lead of the project!

- [x] Before making your PR, refresh your branch with a merge from its parent in case of conflicting changes since you started your branch.
- [x] This PR description is written clearly so that reviewers can understand the why/what/how of the work done.
- [x] Reviewed your own PR and checked your changed code.
- [x] Invited a range of good reviewers.
- [x] Wait for a PR to be properly reviewed. Responded to all questions, and made any required changes.
- [x] Received one other reviewer's approval. Ideally wait for 2+ approvals, especially if you didn't get a thorough reivew on the first approval.
- [x] If a reviewer asks a question, even if you have another approval, got the approval of the other reviewers as well before merging.
- [x] Made sure existing reviewers approved any changes to the PR.
- [x] Resolved any merge conflicts after making the PR.
- [x] CI/CD builds are not reporting any errors.

### CODE REVIEW

File: tools/vol2obj/main.c
Comment (capnramses) [inline tools/vol2obj/main.c:1]: this is the file to review with the critical changes. most of the changes in this PR are copy-pastes of dependencies.

File: tools/vol2obj/main.c [hunk 1]
```diff
@@ -2,54 +2,40 @@
...
  * vol2obj   | Vologram frame to OBJ+image converter.
- * --------- | ----------
- * Version   | 0.7.0
- * Authors   | Anton Gerdelan <anton@volograms.com>
- * Copyright | 2021, Volograms (http://volograms.com/)
- * Language  | C99
+ * --------- | ----------------------------------------------------------------
+ * Version   | 0.8.0
+ * Authors   | Anton Gerdelan  <anton@volograms.com>
+ *           | Jan Ondřej      <jan@volograms.com>
```
Comment (capnramses) [inline tools/vol2obj/main.c:8]: added Jan here since the previous change with no-normals models was by Jan

File: build_msvc.bat
Comment (capnramses) [inline build_msvc.bat:1]: this file updated to pull in dependencies to build

File: Makefile
Comment (capnramses) [inline Makefile:1]: this file updated to pull in dependencies to build

Comment (capnramses) [reply Makefile:1]: also re-organised the bottom section into parts because it got messy after mixing C and C++ (basis stuff) code.

File: README.md
Comment (capnramses) [inline README.md:1]: updated readme. please proof read - i and a typo champion!

**Comment by capnramses** [comment]
i believe i fixed a bug there if you wouldn't mind reviewing again please

File: README.md
Comment (capnramses) [reply README.md:1]: *am

**Comment by patrick-volograms** [comment]
Tested on Mac: 
`make` works perfectly 
Works perfect with single file vologram
Doesn't work with multi-file volograms with basis texture (multi-file always expects video output) (not sure if this is edge case though since we we not be exporting basis volograms as multiple files now) 
Final suggestion: all the usage examples in readme say `vol2obj.exe` but Mac and Linux use `vol2obj.bin`

**Comment by capnramses** [comment]
> Tested on Mac: `make` works perfectly Works perfect with single file vologram Doesn't work with multi-file volograms with basis texture (multi-file always expects video output) (not sure if this is edge case though since we we not be exporting basis volograms as multiple files now) Final suggestion: all the usage examples in readme say `vol2obj.exe` but Mac and Linux use `vol2obj.bin`

thanks!
oh hmm i didn't think multi-file basis texture things existed! i guess there is no harm in supporting those too...
i think i can just put a note on the readme.

**Comment by capnramses** [comment]
> Tested on Mac: `make` works perfectly Works perfect with single file vologram Doesn't work with multi-file volograms with basis texture (multi-file always expects video output) (not sure if this is edge case though since we we not be exporting basis volograms as multiple files now) Final suggestion: all the usage examples in readme say `vol2obj.exe` but Mac and Linux use `vol2obj.bin`

let's put it in the Issues list as an interesting edge case to support. i had a look through the code and it's going to be pretty hairy to implement:

* We would lose the "missing --video parameter" check which is quite handy.
* Drag-and-drop directory support searches for various video filenames. It's hard to know what to prefer by default in terms of 1024, 2048 and integrated textures in this case.
* We would have to do a more comprehensive check for all 3 and add some `is_a_file_and_a_video(x)` type function, then we could stop and tell the user if none are available.

**Review by patrick-volograms — APPROVED** [review]

**Pull request merged**